grails-app/controllers/de/iteratec/osm/measurement/environment/QueueDashboardController.groovy

```diff
@@ -0,0 +1,80 @@
+package de.iteratec.osm.measurement.environment
+import de.iteratec.osm.annotations.RestAction
+import de.iteratec.osm.measurement.schedule.Job
+import de.iteratec.osm.result.JobResult
+import de.iteratec.osm.result.JobResultStatus
+import de.iteratec.osm.result.WptStatus
+import de.iteratec.osm.system.LocationHealthCheck
+import de.iteratec.osm.system.LocationHealthCheckDaoService
+import de.iteratec.osm.util.ControllerUtils
+import grails.validation.Validateable
+
+class QueueDashboardController {
+
+    LocationHealthCheckDaoService locationHealthCheckDaoService
+    QueueAndJobStatusService queueAndJobStatusService
+
+    def index() { }
+
+    def getActiveWptServer()
+    {
+        List<WebPageTestServer> list = WebPageTestServer.findAllByActive(true)
+        return ControllerUtils.sendObjectAsJSON(response, list )
+    }
+
+    @RestAction
+    def getWptServerInformation(Long id)
+    {
+        if(id == null) {
+            return emptyResponse()
+        }
+        WebPageTestServer wptserver = WebPageTestServer.findById(id)
+
+        if(wptserver != null){
+            List<Location> listLocation = Location.findAllByActiveAndWptServer(true, wptserver)
+            List<LocationHealthCheck> healthChecks = locationHealthCheckDaoService.getLatestHealthChecksFor(listLocation)
+            List<Map> listLocationInfo = new ArrayList<Map>()
+
+            listLocation.forEach( {Location location ->
+
+                List<JobResult> executingJobResults
+                executingJobResults = queueAndJobStatusService.getExecutingJobResults(location)
```

Comment:
You can also assign this directly `List<JobResult> executingJobResults = queueAndJob....`